### PR TITLE
Story 489

### PR DESCRIPTION
**Pull request by deepakr123**

### CODE REVIEW

File: migrations/20200513102603-insert_criteria_queries.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      },{
+        query: query,
+        params: [Uuid.random().toString(), 'observation_score_criteria_report_query', '{"queryType":"groupBy","dataSource":"sl_observation_qa","dimensions":["school","schoolName","observationSubmissionId","completedDate","scoreAchieved","observationName","minScore","maxScore","questionName","questionExternalId","questionResponseType","criteriaName","criteriaId"],"aggregations":[{"type":"count","name":"count"}],"granularity":"all","postAggregations":[],"intervals":"1901-01-01T00:00:00+00:00/2101-01-01T00:00:00+00:00","filter":{"type":"and","fields":[{"type":"selector","dimension":"observationId","value":""},{"type":"or","fields":[{"type":"selector","dimension":"questionResponseType","value":"multiselect"},{"type":"selector","dimension":"questionResponseType","value":"radio"},{"type":"selector","dimension":"questionResponseType","value":"slider"}]}]},"limitSpec":{"type":"default","limit":10000,"columns":[{"dimension":"count","direction":"descending"}]}}']
+      }];
+
+      await cassandra.batch(queries, { prepare: true });
+      console.log("successfully inserted the queries");
```
Comment (aks30) [inline migrations/20200513102603-insert_criteria_queries.js]: @deepakr123 - Remove console log

File: migrations/20200513102603-insert_criteria_queries.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      },{
+        query: query,
+        params: [Uuid.random().toString(), 'entity_criteria_report_query', '{"queryType":"groupBy","dataSource":"sl_observation_dev","granularity":"all","dimensions":["completedDate","questionName","questionAnswer","school","schoolName","entityType","observationName","observationId","questionResponseType","questionResponseLabel","observationSubmissionId","questionId","questionExternalId","instanceId","instanceParentQuestion","instanceParentResponsetype","instanceParentId","instanceParentEcmSequence","instanceParentExternalId","criteriaName","criteriaId","instanceParentCriteriaName","instanceParentCriteriaId"],"filter":{"type":"and","fields":[{"type":"selector","dimension":"school","value":""},{"type":"selector","dimension":"observationId","value":""}]},"aggregations":[],"postAggregations":[],"intervals":["1901-01-01T00:00:00+00:00/2101-01-01T00:00:00+00:00"]}']
+      },{
+        query: query,
```
Comment (aks30) [inline migrations/20200513102603-insert_criteria_queries.js:27]: @deepakr123 - Remove limit of 10000

File: views/scoreCriteriaTemplate.ejs [hunk 3]
```diff
@@ -0,0 +1,54 @@
+<html>
+
+<head>
+    <link rel="stylesheet" type="text/css" href="style.css">
+</head>
+
+<body>
+
+    <%if(data.schoolName) {%>
```
Comment (aks30) [inline views/scoreCriteriaTemplate.ejs]: @deepakr123 - Rename it to the entity name

File: helper/common_handler.js [hunk 4]
```diff
@@ -1425,6 +1425,132 @@ async function getScoreChartObject(items) {
...
+        let arrayOfChartData = [];
+
+        await Promise.all(items.map(async element => {
+
+        await Promise.all(element.questionArray.map(async ele => {
+
+            var obj;
+
+            if (ele.chart.type == "pie") {
+
```
Comment (aks30) [inline helper/common_handler.js]: @deepakr123 - Make graph objects as common constants

File: controllers/v1/criteria.js [hunk 5]
```diff
@@ -0,0 +1,1319 @@
...
+const pdfHandler = require('../../helper/common_handler');
+const omit = require('object.omit');
+const url = require("url");
+const rimraf = require("rimraf");
+const fs = require('fs');
+const path = require('path');
+
+
+/**
+   * @api {post} /dhiti/api/v1/criteria/instanceReport
```
Comment (aks30) [inline controllers/v1/criteria.js]: @deepakr123 - Please make the URL as /dhiti/api/v1/observations/instanceReportByCriteria

File: controllers/v1/criteria.js [hunk 6]
```diff
@@ -0,0 +1,1319 @@
...
+
+/**
+   * @api {post} /dhiti/api/v1/criteria/instanceReport 
+   * Instance report
+   * @apiVersion 1.0.0
+   * @apiGroup Criteria
+   * @apiHeader {String} x-auth-token Authenticity token  
+   * @apiParamExample {json} Request-Body:
+* {
+  "submissionId": "",
```
Comment (aks30) [inline controllers/v1/criteria.js]: @deepakr123 - Please add criteria in request body

File: controllers/v1/criteria.js [hunk 7]
```diff
@@ -0,0 +1,1319 @@
...
+                .then(async function (result) {
+
+                    let bodyParam = JSON.parse(result.query);
+
+                    if (config.druid.observation_datasource_name) {
+                        bodyParam.dataSource = config.druid.observation_datasource_name;
+                    }
+
+                    //if filter is given
+                    if (req.body.filter) {
```
Comment (aks30) [inline controllers/v1/criteria.js]: @deepakr123 - Make it as below : -

{
  "submissionId": "",
  "filter": {
      "criteria": [// array of criteria id]
}
}

File: controllers/v1/criteria.js [hunk 8]
```diff
@@ -0,0 +1,1319 @@
...
+      resolve(instaRes);
+    }
+
+  });
+};
+
+
+
+/**
+   * @api {post} /dhiti/api/v1/criteria/instanceScoreReport
```
Comment (aks30) [inline controllers/v1/criteria.js]: @deepakr123 - Move it inside observations/instanceScoreReportByCriteria

File: controllers/v1/criteria.js [hunk 9]
```diff
@@ -0,0 +1,1319 @@
...
+
+/**
+   * @api {post} /dhiti/api/v1/criteria/instanceScoreReport 
+   * Instance score report
+   * @apiVersion 1.0.0
+   * @apiGroup Criteria
+   * @apiHeader {String} x-auth-token Authenticity token  
+   * @apiParamExample {json} Request-Body:
+* {
+  "submissionId": "",
```
Comment (aks30) [inline controllers/v1/criteria.js]: @deepakr123 - Add filter in request body

File: controllers/v1/criteria.js [hunk 10]
```diff
@@ -0,0 +1,1319 @@
...
+                  message: "INTERNAL_SERVER_ERROR"
+              };
+              resolve(response);
+          });
+  })
+}
+
+
+//Function to make a call to samiksha assessment entities list API
+async function getTotalSchools(observationId,token) {
```
Comment (aks30) [inline controllers/v1/criteria.js]: @deepakr123 - Lets not use school

File: controllers/v1/criteria.js (hunk 7 above)
Comment (deepakr123) [reply controllers/v1/criteria.js]: done

File: migrations/20200513102603-insert_criteria_queries.js (hunk 1 above)
Comment (deepakr123) [reply migrations/20200513102603-insert_criteria_queries.js]: done

File: controllers/v1/criteria.js (hunk 9 above)
Comment (deepakr123) [reply controllers/v1/criteria.js]: done

File: helper/common_handler.js (hunk 4 above)
Comment (deepakr123) [reply helper/common_handler.js]: done

File: controllers/v1/criteria.js (hunk 6 above)
Comment (deepakr123) [reply controllers/v1/criteria.js]: done

File: controllers/v1/criteria.js (hunk 10 above)
Comment (deepakr123) [reply controllers/v1/criteria.js]: done

File: migrations/20200513102603-insert_criteria_queries.js (hunk 2 above)
Comment (deepakr123) [reply migrations/20200513102603-insert_criteria_queries.js:27]: done

File: views/scoreCriteriaTemplate.ejs (hunk 3 above)
Comment (deepakr123) [reply views/scoreCriteriaTemplate.ejs]: done

**Pull request merged**